server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java

```diff
@@ -508,6 +525,14 @@ private static boolean seedsChanged(final List<String> oldSeedNodes, final List<
         return oldSeeds.equals(newSeeds) == false;
     }
 
+    private static boolean authorizationChanged(String oldOptionalCredential, String newOptionalCredential) {
```

Comment:
`oldOptionalAuthorization` and `newOptionalAuthorization`? or just `old` and `current`?

Reply:
I copied the existing proxyChanged method are changed `proxy` to `authorization`. Is that OK?